server/src/main/java/io/crate/analyze/SymbolEvaluator.java

```diff
@@ -72,7 +73,17 @@ public static Object evaluate(TransactionContext txnCtx,
     @Override
     public Input<?> visitParameterSymbol(ParameterSymbol parameterSymbol, Row params) {
         return () -> {
-            Object value = params.get(parameterSymbol.index());
+            Object value;
+            try {
+                 value = params.get(parameterSymbol.index());
+            } catch (IndexOutOfBoundsException e) {
+                throw new IllegalArgumentException(String.format(
+                    Locale.ENGLISH,
+                    "The query contains a parameter placeholder $%d, but there are only %d parameter values",
+                    (parameterSymbol.index() + 1),
+                    params.numColumns()
+                ));
+            }
```

Comment:
Can't we re-use code from SubQueryAndParamBinder.convert? 
They are not completely same so maybe we could take out some common parts into separate static method.

Reply:
I tried, but I cannot use the same logic for the second part (casting).
So I extracted only the index binding to `ParameterSymbol`.
Thx for the suggestion.